### PR TITLE
Ignore public method calls when destroying; throw when destroyed

### PR DESCRIPTION
**Pull request by feross**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

When we call public methods like `signal()` but the peer is already in a `destroying` state, we might get an exception. For example, if the `RTCPeerConnection` constructor throws, we enter `destroying` but there's no way that the user can know that yet, since none of the destroy events have fired.

So, I propose that we silently disregard calls to public API methods when we're in the `destroying` state since destroy will happen on the next tick anyway. Furthermore, we should actually throw when any of these methods are called in the `destroyed` state, which were actually doing for `signal()` but not for any of the other methods. I named this error type `ERR_DESTROYED` for all methods.

**Which issue (if any) does this pull request address?**

This fixes this use case in `bittorrent-tracker`:

```js
const peer = new Peer()
peer.signal()
```

You can see an example stack trace in this issue: https://github.com/webtorrent/webtorrent-desktop/issues/1912

If the peer constructor throws then `this._pc` will be undefined and there's no way for us to know that. We *could* attempt to handle it by checking `this.destroying` but that actually doesn't work because there's an extra `queueMicrotask` before this.destroy() is even called. This was a mistake leftover from the introduction of `this.destroying` in this PR: https://github.com/feross/simple-peer/pull/723 We only need one `queueMicrotask` and we can just use the one in `destroy()`. This PR removes the extra call.

We don't actually want to force the user to check `this.destroying` before every call to a public API method, so we handle it internally.

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -180,7 +180,8 @@ class Peer extends stream.Duplex {
   }
 
   signal (data) {
-    if (this.destroyed) throw errCode(new Error('cannot signal after peer is destroyed'), 'ERR_SIGNALING')
+    if (this.destroying) return
+    if (this.destroyed) throw errCode(new Error('cannot signal after peer is destroyed'), 'ERR_DESTROYED')
```
Comment (jhiesey) [inline index.js:184]: I don't like how there are now three behaviors for all public methods:
* Before `destroy` is called, calling a method has the desired effect
* Once `destroying` is set but before `destroyed` is set, calling a method does nothing
* Once `destroyed` is set it throws

Why not throw when either `destroying` or `destroyed` is set?

**Review by jhiesey — COMMENTED** [review]
See comment

File: index.js (hunk 1 above)
Comment (jhiesey) [reply index.js:184]: On second thought, I can't think of any better options without breaking backwards compatibility.

At least the `error` event gets emitted, so the client should be cleaning everything up.

**Review by jhiesey — APPROVED** [review]
I think this is the best solution without breaking backwards compatibility.

**Comment by feross** [comment]
9.10.0

**Pull request merged**